exchanges/okx/okx_websocket.go

```diff
@@ -0,0 +1,2155 @@
+			Bid:          response.Data[i].BidPrice,
+			Ask:          response.Data[i].BestAskPrice,
+			BidSize:      response.Data[i].BidSize,
+			AskSize:      response.Data[i].BestAskSize,
+			Last:         response.Data[i].LastTradePrice,
+			AssetType:    a,
+			Pair:         c,
+			LastUpdated:  response.Data[i].TickerDataGenerationTime,
+		}
+	}
+	return nil
+}
+
+// GenerateDefaultSubscriptions returns a list of default subscription message.
+func (ok *Okx) GenerateDefaultSubscriptions() ([]stream.ChannelSubscription, error) {
+	var subscriptions []stream.ChannelSubscription
+	assets := ok.GetAssetTypes(true)
+	if ok.Websocket.CanUseAuthenticatedEndpoints() {
+		defaultSubscribedChannels = append(defaultSubscribedChannels,
+			okxChannelAccount,
+			okxChannelOrders,
+		)
+	}
+	for x := range assets {
+		pairs, err := ok.GetEnabledPairs(assets[x])
+		if err != nil {
+			return nil, err
+		}
+		for y := range defaultSubscribedChannels {
+			if defaultSubscribedChannels[y] == okxChannelCandle5m ||
+				defaultSubscribedChannels[y] == okxChannelTickers ||
+				defaultSubscribedChannels[y] == okxChannelOrders ||
+				defaultSubscribedChannels[y] == okxChannelOrderBooks ||
+				defaultSubscribedChannels[y] == okxChannelOrderBooks5 ||
+				defaultSubscribedChannels[y] == okxChannelOrderBooks50TBT ||
+				defaultSubscribedChannels[y] == okxChannelOrderBooksTBT ||
+				defaultSubscribedChannels[y] == okxChannelTrades {
+				for p := range pairs {
+					subscriptions = append(subscriptions, stream.ChannelSubscription{
+						Channel:  defaultSubscribedChannels[y],
+						Asset:    assets[x],
+						Currency: pairs[p],
+					})
+				}
+			} else {
+				subscriptions = append(subscriptions, stream.ChannelSubscription{
+					Channel: defaultSubscribedChannels[y],
+				})
+			}
+		}
+	}
+	if len(subscriptions) >= 240 {
+		log.Warnf(log.WebsocketMgr, "OKx has 240 subscription limit, only subscribing within limit. Requested %v", len(subscriptions))
+		subscriptions = subscriptions[:239]
+	}
+	return subscriptions, nil
+}
+
+// wsProcessPushData processes push data coming through the websocket channel
+func (ok *Okx) wsProcessPushData(data []byte, resp interface{}) error {
+	if err := json.Unmarshal(data, resp); err != nil {
+		return err
+	}
+	ok.Websocket.DataHandler <- resp
+	return nil
+}
+
+// Websocket Trade methods
+
+// WsPlaceOrder places an order thought the websocket connection stream, and returns a SubmitResponse and error message.
+func (ok *Okx) WsPlaceOrder(arg *PlaceOrderRequestParam) (*OrderData, error) {
+	if arg == nil {
+		return nil, errNilArgument
+	}
+	if arg.InstrumentID == "" {
+		return nil, errMissingInstrumentID
+	}
+	if !(strings.EqualFold(TradeModeCross, arg.TradeMode) || strings.EqualFold(TradeModeIsolated, arg.TradeMode) || strings.EqualFold(TradeModeCash, arg.TradeMode)) {
+		return nil, errInvalidTradeModeValue
+	}
+	if !(strings.EqualFold(arg.Side, "buy") || strings.EqualFold(arg.Side, "sell")) {
```

Comment:
```suggestion
	if !strings.EqualFold(arg.Side, "buy") && !strings.EqualFold(arg.Side, "sell") {
```
Do these need to be equalfold?